scalafix/rules/src/main/scala/rsc/rules/semantics/Scopes.scala

```diff
@@ -14,7 +15,14 @@ sealed trait Scope {
 
   def getRename(name: n.Name): String = ""
 
-  protected def member(symtab: Symtab, sym: String, name: n.Name): String = {
+  private val _cache = new java.util.HashMap[MemberKey, String]
+
+  protected def member(symtab: Symtab, sym: String, name: n.Name): String =
+    _cache.computeIfAbsent(MemberKey(symtab, sym, name), _member)
```

Comment:
Since you do a lot of cacheing, how about using a function like this:

```
  /**
   * Memoize the results of a function call to prevent repeated calls to disk
   * This is a short lived application so this should be a safe operation
   */
  def memoize[I, O](f: I => O): I => O = new mutable.HashMap[I, O]() {
    override def apply(key: I) = getOrElseUpdate(key, f(key))
  }
```